skbio/metadata/_metadata.py

```diff
@@ -0,0 +1,1390 @@
+        return not (self == other)
+
+    def to_dataframe(self, encode_missing=False):
+        """Create a pandas dataframe from the metadata.
+
+        The dataframe's index name (``Index.name``) will match this metadata
+        object's ``id_header``, and the index will contain this metadata
+        object's IDs. The dataframe's column names will match the column names
+        in this metadata. Categorical columns will be stored as
+        ``dtype=object`` (containing strings), and numeric columns will be
+        stored as ``dtype=float``.
+
+        Parameters
+        ----------
+        encode_missing : bool, optional
+            Whether to convert missing values (NaNs) back into their original
+            vocabulary (strings) if a missing scheme was used.
+
+        Returns
+        -------
+        pandas.DataFrame
+            Dataframe constructed from the metadata.
+
+        """
+        df = self._dataframe.copy()
+        if encode_missing:
+
+            def replace_nan(series):
+                missing = _missing.series_extract_missing(series)
+                # avoid dtype changing if there's no missing values
+                if not missing.empty:
+                    series[missing.index] = missing
+                return series
+
+            df = df.apply(replace_nan)
+
+        return df
+
+    def get_column(self, name):
+        """Retrieve metadata column based on column name.
+
+        Parameters
+        ----------
+        name : str
+            Name of the metadata column to retrieve.
+
+        Returns
+        -------
+        MetadataColumn
+            Requested metadata column (``CategoricalMetadataColumn`` or
+            ``NumericMetadataColumn``).
+
+        See Also
+        --------
+        get_ids
+
+        """
+        try:
+            series = self._dataframe[name]
+            missing_scheme = self._columns[name].missing_scheme
+        except KeyError:
+            raise ValueError(
+                "%r is not a column in the metadata. Available columns: "
+                "%s" % (name, ", ".join(repr(c) for c in self.columns))
+            )
+
+        return self._metadata_column_factory(series, missing_scheme)
+
+    def get_ids(self, where=None):
+        """Retrieve IDs matching search criteria.
+
+        Parameters
+        ----------
+        where : str, optional
+            SQLite WHERE clause specifying criteria IDs must meet to be
+            included in the results. All IDs are included by default.
+
+        Returns
+        -------
+        set
+            IDs matching search criteria specified in `where`.
+
+        See Also
+        --------
+        ids
+        filter_ids
+        get_column
+
+        Notes
+        -----
+        The ID header (``Metadata.id_header``) may be used in the `where`
+        clause to query the table's ID column.
+
+        """
+        if where is None:
+            return set(self._ids)
+
+        conn = sqlite3.connect(":memory:")
```

Comment:
It looks like this is loaded in memory by default.  For future iterations, we may want to consider offloading to disk, particularly when we are dealing with large databases.